### ISSUE TITLE
Empty master password is wrong until a wrong password is entered

### ISSUE THREAD

**Issue by seatedscribe** [keepassxreboot/keepassxc#570]
## Context
This is a substitute for #569.
I dug a little into the issue and found a more generic scenario:

whenever a DB is saved and its master password is empty, in order to unlock it is needed to provide a wrong password first.

Describing the original issue I was misleaded into thinking this had to do with "saved as" DBs only as I had used a lot of empty passwords in the past and never had any trouble.

So I am suspecting this is something related to fresh code.

## Steps to Reproduce (for bugs)

1. Create a new Database with empty password
2. (optional) add an entry to the database
3. Save the database
4. Lock
5. Try to unlock with empty password, this will throw an error
6. Entry a wrong password
7. Try to unlock with empty password, this time you unlock sucessfully

## Your Environment

* KeePassXC version/commit used: 2.1.4
* Qt version (e.g. Qt 5.3): 5.5.1

**Comment by TheZ3ro** [comment]
Ok, confirmed. I'm editing the first comment with step to reproduce

**Comment by phoerious** [comment]
Does "empty password" mean "empty password" or "no password"? Those are two different things.

**Comment by TheZ3ro** [comment]
Thanks to @phoerious comment I've find out what's the "problem".
Some PRs ago we explicitly allowed the creation of database encrypted with **empty string as password** with a dialog asking *"Do you really want to use an empty string as password?"* and preventing user from creating DB without a password (**no password**)

When you unlock a DB with an **empty password**, by default no locking mechanism is selected (password, keyfile or yubikey). So if you press enter or click "Ok" to unlock, since no method is selected, an error will be printed.
By starting to type the password method will be selected. You type a wrong password, the password method will remain selected with no text in the textbox, thus allowing you to unlock the DB.

TL;DR. If you are using an empty password, you still need to select the password checkbox when unlocking.

I think this issue is invalid, alternatively we can think of enabling the password method by default since users can't create a DB without a password.

Creating a DB with **no password**
![istantanea_2017-05-14_20-48-37](https://cloud.githubusercontent.com/assets/686894/26036878/c5b8a434-38e6-11e7-9a1c-cd5799c8d120.png)

**Comment by phoerious** [comment]
Users can't create a DB with only a YubiKey, but I guess that's rather a bug.

**Comment by TheZ3ro** [comment]
Yup, it's a (different) bug.  
You can't select only keyfile or only yubikey without selecting the password as well.
This was the PR/commit https://github.com/keepassxreboot/keepassxc/commit/2d4e08e060ce18d0040ffcc0c9a2af17e28a81ff

**Comment by seatedscribe** [comment]
I'm totally for enabling the password checkbox by default.

Good job @TheZ3ro!

P.S. I know that this could be seen as a non-qualified issue report, since
a user should know the difference between empty and nonexistent, before
reporting an "invalid" problem. This is basic computer theory but I wanted
to wear the hat of the ordinary "man on the street" here, who will promptly
reply with a "no differences, dude!" if asked on the topic.

**Comment by droidmonkey** [comment]
This issue does not exist in 2.3.1 anymore. Although the password checkbox is unchecked when reopening a database that was saved without a password which causes an HMAC error. When the checkbox is checked everything works fine.
